### PR TITLE
Add detection of Azure App Service to CloudPlatform

### PR DESCRIPTION
**Pull request by jdubois**
__Add Azure platform support to Spring Boot.__

More specifically, this adds "Azure App Service" support to Spring Boot - as this is the Azure standard PaaS service, it makes sense to give it the "AZURE" enum name. If it provides any additional value, we can add later Azure Functions and Azure Spring Cloud, but those shouldn't be the default and would have their own enumeration name, as they work very differently.

This discovers Azure using [the "WEBSITE_SITE_NAME" environment properties](https://stackoverflow.com/questions/25678419/how-to-check-if-code-is-running-on-azure-websites), which is similar to what is done with Heroku (using the "DYNO" environment property).

I put Azure near the top of the file, trying to have some alphabetical sorting after "NONE", as I believe this is how people will read that class.

As a consequence, setting this value should put `server.forward-headers-strategy=native` by default, which will allow applications to receive sensitive security headers from the reverse proxy. More specifically, this will solve this error, when using Active Directory to secure App Service, which looks pretty common: https://stackoverflow.com/questions/60365017/redirect-url-for-spring-oauth2-app-on-azure-with-active-directory-invalid-redir

### CODE REVIEW

**Comment by jdubois** [comment]
After internal discussion at Microsoft, I'm going to modify this PR to use "AZURE_APP_SERVICE" instead of "AZURE", in order to be clear on the service that is being used.

**Comment by cuspymd** [comment]
There seems to be no problem with the operation, but I'm afraid I can't understand it. It's a very common variable name, could you be sure that only azure uses this variable name?

Is it difficult to add an environment variable representing azure? It may sound rude, but I ask because I'm really curious.

**Comment by jdubois** [comment]
You are correct @cuspymd but unfortunately there isn't a good specific environment variable for Azure. Let me try to find a second one, and then I'll use the combinaison of them to reduce the risk.

**Comment by jdubois** [comment]
I pushed my commit, but I know there is a formatting issue - it breaks when I "gradle build" on my machine, but it doesn't give me any indication of what the formatting rules are, or what the error is.... I'm hoping I'll have a better idea with the CI build.

**Comment by cuspymd** [comment]
@jdubois Thank you for checking.

File: spring-boot-project/spring-boot/src/main/java/org/springframework/boot/cloud/CloudPlatform.java [hunk 1]
```diff
@@ -17,11 +17,7 @@
 package org.springframework.boot.cloud;
 
 import org.springframework.boot.context.properties.bind.Binder;
-import org.springframework.core.env.ConfigurableEnvironment;
-import org.springframework.core.env.EnumerablePropertySource;
-import org.springframework.core.env.Environment;
-import org.springframework.core.env.PropertySource;
-import org.springframework.core.env.StandardEnvironment;
+import org.springframework.core.env.*;
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot/src/main/java/org/springframework/boot/cloud/CloudPlatform.java:20]: Can you please revert this change to the imports?

File: spring-boot-project/spring-boot/src/test/java/org/springframework/boot/cloud/CloudPlatformTests.java [hunk 2]
```diff
@@ -50,7 +50,17 @@ void getActiveWhenNotInCloudShouldReturnNull() {
 		Environment environment = new MockEnvironment();
 		CloudPlatform platform = CloudPlatform.getActive(environment);
 		assertThat(platform).isNull();
+	}
 
+	@Test
+	void getActiveWhenHasWebsiteSiteNameShouldReturnAzureAppService() {
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot/src/test/java/org/springframework/boot/cloud/CloudPlatformTests.java:56]: It would be good to have a test or two that asserts that `AZURE_APP_SERVICE` isn't active when only one of the environment variables is set.

**Review by wilkinsona — CHANGES_REQUESTED** [review]
Thanks for the PR, @jdubois. I've left a couple of comments for your consideration.

File: spring-boot-project/spring-boot/src/test/java/org/springframework/boot/cloud/CloudPlatformTests.java (hunk 2 above)
Comment (jdubois) [reply spring-boot-project/spring-boot/src/test/java/org/springframework/boot/cloud/CloudPlatformTests.java:56]: Thank you for your review, I have added 2 tests

**Review by jdubois — COMMENTED** [review]
I fixed both issues, @wilkinsona can you check and approve the review?

**Review by wilkinsona — APPROVED** [review]

**Comment by wilkinsona** [comment]
Thanks very much, @jdubois. The proposed changes have now been merged.
